src/Features/VisualBasic/Portable/Completion/CompletionProviders/EnumCompletionProvider.vb

```diff
@@ -31,77 +31,47 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
-                    Return m.Kind = SymbolKind.Field AndAlso
-                        DirectCast(m, IFieldSymbol).IsConst AndAlso
-                        m.IsEditorBrowsable(options.HideAdvancedMembers, context.SemanticModel.Compilation)
-                End Function).ToImmutableArray()
-
-            Return Task.FromResult(result)
-        End Function
-
-        Private Shared Function GetNormalSymbolsAsync(
-                context As VisualBasicSyntaxContext, position As Integer, options As CompletionOptions, cancellationToken As CancellationToken) As Task(Of ImmutableArray(Of ISymbol))
+        Protected Overrides Function GetSymbolsAsync(
+                completionContext As CompletionContext,
+                syntaxContext As VisualBasicSyntaxContext,
+                position As Integer,
+                options As CompletionOptions,
+                cancellationToken As CancellationToken) As Task(Of ImmutableArray(Of SymbolAndSelectionInfo))
 
-            If context.SyntaxTree.IsInNonUserCode(context.Position, cancellationToken) OrElse
-                context.SyntaxTree.IsInSkippedText(position, cancellationToken) Then
-                Return SpecializedTasks.EmptyImmutableArray(Of ISymbol)()
-            End If
+            Dim builder = ArrayBuilder(Of SymbolAndSelectionInfo).GetInstance()
+            Try
 
-            If context.TargetToken.IsKind(SyntaxKind.DotToken) Then
-                Return SpecializedTasks.EmptyImmutableArray(Of ISymbol)()
-            End If
+                If syntaxContext.SyntaxTree.IsInNonUserCode(syntaxContext.Position, cancellationToken) Then
+                    Return SpecializedTasks.EmptyImmutableArray(Of SymbolAndSelectionInfo)()
+                End If
 
-            Dim typeInferenceService = context.GetLanguageService(Of ITypeInferenceService)()
-            Dim span = New TextSpan(position, 0)
-            Dim enumType = typeInferenceService.InferType(context.SemanticModel, position, objectAsDefault:=True, cancellationToken:=cancellationToken)
+                ' This providers provides fully qualified names, eg "DayOfWeek.Monday"
+                ' Don't run after dot because SymbolCompletionProvider will provide
+                ' members in situations like Dim x = DayOfWeek.$$
+                If syntaxContext.TargetToken.IsKind(SyntaxKind.DotToken) Then
+                    Return SpecializedTasks.EmptyImmutableArray(Of SymbolAndSelectionInfo)()
+                End If
 
-            If enumType.TypeKind <> TypeKind.Enum Then
-                Return SpecializedTasks.EmptyImmutableArray(Of ISymbol)()
-            End If
+                Dim typeInferenceService = syntaxContext.GetLanguageService(Of ITypeInferenceService)()
+                Dim enumType = typeInferenceService.InferType(syntaxContext.SemanticModel, position, objectAsDefault:=True, cancellationToken:=cancellationToken)
 
-            Dim otherSymbols = context.SemanticModel.LookupSymbols(position).WhereAsArray(
```

Comment:
Not sure why this is here, Seems to be redundant since those symbols should be provided by SymbolCompletionProvider